integration-cli/docker_cli_network_unix_test.go

```diff
@@ -1417,8 +1418,18 @@ func (s *DockerNetworkSuite) TestDockerNetworkInternalMode(c *check.C) {
 	c.Assert(waitRun("first"), check.IsNil)
 	dockerCmd(c, "run", "-d", "--net=internal", "--name=second", "busybox", "top")
 	c.Assert(waitRun("second"), check.IsNil)
-	_, _, err := dockerCmdWithError("exec", "first", "ping", "-c", "1", "www.google.com")
-	c.Assert(err, check.NotNil)
-	_, _, err = dockerCmdWithError("exec", "second", "ping", "-c", "1", "first")
+	errchan := make(chan error)
+	cmd := exec.Command(dockerBinary, "exec", "first", "ping", "-c", "1", "www.google.com")
+	go func() {
+		errchan <- cmd.Run()
+		close(errchan)
+	}()
+	select {
+	case <-time.After(time.Second):
+		cmd.Process.Kill()
```

Comment:
I don't think use a timeout here is a good idea. This test is to make sure we can't access to external network with `internal` network, `exec.Command(dockerBinary, "exec", "first", "ping", "-c", "1", "www.google.com")` should always return error,  has a timeout here will break this test,  if this command will success after few seconds and this test should suppose to be failed, but it has been killed by timeout and we miss the error. :)